server.go

```diff
@@ -220,6 +220,11 @@ func (s *Server) Close() error {
 		return ErrServerClosed
 	}
 
+	// Stop goroutines.
+	if s.done != nil {
+		close(s.done)
+	}
+
```

Comment:
I don't know if that's the blowup you're seeing but could be.  The command processor loop could (easily) be in the middle of a command.